plugin/cmd/kube-scheduler/app/configurator.go

```diff
@@ -186,3 +191,113 @@ func (sc schedulerConfigurator) Create() (*scheduler.Config, error) {
 
 	return sc.CreateFromConfig(*policy)
 }
+
+func (sc *schedulerConfigurator) SetupPolicyConfigMapEventHandlers(client clientset.Interface, informerFactory informers.SharedInformerFactory) {
+	// selector targets only the scheduler's policy ConfigMap.
+	selector := cache.NewListWatchFromClient(client.CoreV1().RESTClient(), "configmaps", sc.policyConfigMapNamespace, fields.OneTermEqualSelector(api.ObjectNameField, string(sc.policyConfigMap)))
+
+	sharedIndexInformer := informerFactory.InformerFor(&v1.ConfigMap{}, func(client clientset.Interface, resyncPeriod time.Duration) cache.SharedIndexInformer {
+		sharedIndexInformer := cache.NewSharedIndexInformer(
+			selector,
+			&v1.ConfigMap{},
+			resyncPeriod,
+			cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		)
+		return sharedIndexInformer
+	})
+
+	sharedIndexInformer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs {
+			AddFunc:    nil, // scheduler aborts if its policy ConfigMap does not exist. So, we do not need and "add" handler.
+			UpdateFunc: sc.updatePolicyConfigMap,
+			DeleteFunc: sc.deletePolicyConfigMap,
+		},
+		0,
+	)
+}
+
+// verifyNewPolicyConfig verifies that the new object received by the ConfigMap watch is a valid policy config.
+func verifyNewPolicyConfig(obj interface{}) error {
+	newConfig, ok := obj.(*v1.ConfigMap)
+	if !ok {
+		return fmt.Errorf("cannot convert obj to *v1.ConfigMap: %v", obj)
+	}
+	newPolicy, ok := newConfig.Data[options.SchedulerPolicyConfigMapKey]
+	if !ok {
+		return fmt.Errorf("No element with key = '%v' is found in the ConfigMap.Data", options.SchedulerPolicyConfigMapKey)
+	}
+	configData := []byte(newPolicy)
+	var policy schedulerapi.Policy
+	if err := runtime.DecodeInto(latestschedulerapi.Codec, configData, &policy); err != nil {
+		return fmt.Errorf("invalid scheduler policy configuration: %v", err)
+	}
+	// validate the policy configuration
+	if err := validation.ValidatePolicy(policy); err != nil {
+		return fmt.Errorf("The new scheduler policy is invalid. Will keep using the old policy! Error: %v", err)
+	}
+	return nil
+}
+
+func (sc *schedulerConfigurator) addPolicyConfigMap(obj interface{}) {
+	glog.Info("Received a request to add a scheduler policy config.")
```

Comment:
Please make sure the policy is logged on scheduler startup. (I didn't check; it may be already.)